x/multisig/decorator_test.go

```diff
@@ -176,3 +165,18 @@ var _ weave.Tx = ContractTx{}
 func (p ContractTx) GetMultisig() [][]byte {
 	return p.MultisigID
 }
+
+func createContract(t *testing.T, db weave.KVStore, msg CreateContractMsg) []byte {
+	k := weavetest.NewCondition()
```

Comment:
Bit confused why we need k here.

Reply:
I do not fully understand this as this was here before :see_no_evil: 

If I remove the registration of a random condition for the purpose of authentication one test is failing:

```
--- FAIL: TestDecorator (0.00s)
    require.go:794: 
                Error Trace:    decorator_test.go:178
                                                        decorator_test.go:27
                Error:          Received unexpected error:
                            
                                github.com/iov-one/weave/x/multisig.CreateContractMsgHandler.validate
                                        /home/piotr/go/src/github.com/iov-one/weave/x/multisig/handlers.go:67
                                github.com/iov-one/weave/x/multisig.CreateContractMsgHandler.Deliver
                                        /home/piotr/go/src/github.com/iov-one/weave/x/multisig/handlers.go:43
                                github.com/iov-one/weave/x/multisig.createContract
                                        /home/piotr/go/src/github.com/iov-one/weave/x/multisig/decorator_test.go:177
                                github.com/iov-one/weave/x/multisig.TestDecorator
                                        /home/piotr/go/src/github.com/iov-one/weave/x/multisig/decorator_test.go:27
                                No signer: unauthorized
                Test:           TestDecorator
                Messages:       cannot deliver
FAIL
```

Without looking into details I think this might be because there is no signature authentication is failing without even trying. I will take time to understand this code :nerd_face: 
Is this an expected behaviour?

Reply:
After I have realized what it was doing, the code was updated. As @alpe pointed out in https://github.com/iov-one/weave/pull/416#discussion_r267768537 it can be done much better.